File: frontend/src/merchant/object/profile/UpdateMerchantProfileRequest.ts
import type { PhoneNumber } from '@/shared/object/domain/DomainObjects'
import type { MerchantPayoutAccount } from '@/merchant/object/profile/MerchantPayoutAccount'

export type UpdateMerchantProfileRequest = {
  contactPhone: PhoneNumber
  payoutAccount: MerchantPayoutAccount
}

File: backend/src/main/scala/domain/merchant/UpdateMerchantProfileRequest.scala
package domain.merchant

import domain.shared.given

import domain.shared.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class UpdateMerchantProfileRequest(
    contactPhone: PhoneNumber,
    payoutAccount: MerchantPayoutAccount,
)
object UpdateMerchantProfileRequest:
  given Encoder[UpdateMerchantProfileRequest] = deriveEncoder
  given Decoder[UpdateMerchantProfileRequest] = deriveDecoder
